RaspiXUD.md

```diff
@@ -0,0 +1,183 @@
+# open the file to edit
+sudo nano ./etc/netplan/wireless.yaml
+# strip down the file to the bare minmum for the Pi to get an IP automatically assigned by your router
+network:
+  version: 2
+  wifis:
+    wlan0:
+      dhcp4: yes
+      dhcp6: no
+      access-points:
+        "<YOUR WIFI SSID>":
+          password: "<YOUR WIFI PASSWORD>"
+# if you can't access your router to read out your Pi's IP, you can also configure a static IP now
+      addresses: [192.168.1.42/24]
+      gateway4: 192.168.1.1
+      nameservers:
+        addresses: [192.168.1.1, 8.8.8.8]
+# CTRL+S, CTRL+X.
+```
+4. Insert the microSD card into your Pi, connect it to your router via ethernet cable and to a power supply. Connecting a screen via HDMI and a USB keyboard makes life easier, but checking the assigned IP in your router and SSHing in from your computer works too.
+5. Follow the inital setup instructions. Default user + password is `ubuntu`. You will be asked to change the password on first login.
+6. Update ubuntu via `sudo apt update && sudo apt upgrade`
+7. Install docker following the [official instructions](https://docs.docker.com/install/linux/docker-ce/ubuntu/) (select `arm64` in step 4 of "Set up the repository"). At the time of writing, ubuntu 19.10 was still not supported in the official docker repos. If you see an error along the lines `'docker-ce' has no installation candidate`, run this to set up the repository instead:
+```
+sudo add-apt-repository \
+   "deb [arch=arm64] https://download.docker.com/linux/ubuntu \
+   disco \
+   stable"
+```
+8. Add new user `xud`:
+
+```bash
+ubuntu@ubuntu:~$ sudo adduser xud
+Adding user `xud' ...
+Adding new group `xud' (1001) ...
+Adding new user `xud' (1001) with group `xud' ...
+Creating home directory `/home/xud' ...
+Copying files from `/etc/skel' ...
+New password: 
+Retype new password: 
+passwd: password updated successfully
+Changing the user information for xud
+Enter the new value, or press ENTER for the default
+	Full Name []: 
+	Room Number []: 
+	Work Phone []: 
+	Home Phone []: 
+	Other []: 
+
+Is the information correct? [Y/n] ubuntu@ubuntu:~$ Y
+```
+9. Add the `xud` user to the sudo group (advanced users can skip this and use another user to run sudo commands), the docker group and test if docker is working:
+```bash
+ubuntu@ubuntu:~$ sudo usermod -aG sudo xud
+ubuntu@ubuntu:~$ sudo usermod -aG docker xud
+# switch to user xud
+ubuntu@ubuntu:~$ sudo su - xud
+xud@ubuntu:~$ docker run hello-world
+Hello from Docker!
+This message shows that your installation appears to be working correctly.
+```
+10. Looking good! Optionally, add an alias to enter your xud environment by simply typing "xud":
+```bash
+xud@ubuntu:~$ sudo nano ~/.bash_aliases
+# add the line
+alias xud='bash ~/xud.sh'
+# CTRL+S, CTRL+X. Then run
+xud@ubuntu:~$ source ~/.bashrc
+```
+11. Connect an ext4-formatted USB stick to your Pi and set it up. It is very important to do this (given you do not want to loose money)!
```

Comment:
Does the filesystem have to be ext4 only? The USB stick could be FAT/exFAT format for some users becuase the want to access the stick content on their Windows or macOS.

Reply:
True, removed that.